src/Microsoft.ML.Data/Training/ITrainerEstimator.cs

```diff
@@ -8,7 +8,6 @@ namespace Microsoft.ML.Runtime.Training
 {
     public interface ITrainerEstimator<out TTransformer, out TPredictor>: IEstimator<TTransformer>
         where TTransformer: ISingleFeaturePredictionTransformer<TPredictor>
-        where TPredictor: IPredictor
```

Comment:
>IPredictor [](start = 26, length = 10)

Hmmm. Is removing this necessary?

I guess your goal is to have the calibrator estimator fitting be a trainer *estimator*, but is that required or helpful, in some fashion?

Reply:
TPredictor is an ICalibrator, for the CalibratorEstimators. ICalibrator is not an IPredictor. 

---
In reply to: [238802348](https://github.com/dotnet/machinelearning/pull/1816#discussion_r238802348) [](ancestors = 238802348)

Reply:
OK. I think that we should then probably then not have this be a training estimator?

Reply:
Pete seemed to like the idea. He was ok with removing this constraint. 

Reverted the change. 

---
In reply to: [239170299](https://github.com/dotnet/machinelearning/pull/1816#discussion_r239170299) [](ancestors = 239170299)